src/java.base/share/native/libverify/check_code.c

```diff
@@ -468,7 +468,8 @@ static void CCout_of_memory (context_type *);
 /* Because we can longjmp any time, we need to be very careful about
  * remembering what needs to be freed. */
 
-static void check_and_push(context_type *context, const void *ptr, int kind);
+static void check_and_push(context_type *context, void *ptr, int kind);
+static void check_and_push_const(context_type *context, const void *ptr, int kind);
```

Comment:
This really needs a comment explaining why two functions are needed.

Reply:
I think it would be clearer to drop the kind argument and have two functions
check_and_push_vm_string(context_type*, const char*)
check_and_push_malloc_block(context_type*, void*)
with a shared helper that is given the appropriate kind.

Reply:
I had that at some point and then dropped it. I don't know why and it does seem like the better option so I'll try it again.